WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift

```diff
@@ -0,0 +1,77 @@
+import UIKit
+
+class JetpackBannerView: UIView {
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    override var intrinsicContentSize: CGSize {
+        return isHidden ? CGSize.zero : super.intrinsicContentSize
+    }
+
+    func setup() {
+
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = Appearance.jetpackBackgroundColor
+
+        let jetpackButton = makeJetpackButton()
+        addSubview(jetpackButton)
+
+        pinSubviewToAllEdges(jetpackButton)
+        heightAnchor.constraint(equalToConstant: Appearance.jetpackBannerHeight).isActive = true
```

Comment:
As previously observed [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18978#discussion_r912561395), setting this constraint in here forces the banner to have the same height everywhere. This is probably fine, but will be a limit if, for any reason, we want different heights. Would it be possible to extrapolate this constraint and set it on the view(s) that hosts the banner instead?

Reply:
We could also expose an overridable `height` property and set the constraint in the view: this would have the advantage of setting a default height, and also not allow to go below a minimum value (since we have the constraints of the button image in there too).

Reply:
Or, if we want to avoid setting absolute sizes in the view at all, we might assign the button constraints proportionally? Whichever you think is best

Reply:
That's a very good point @Gio2018!

I removed the height constraint and now the banner respects the height constraint added by the host.

I felt it's more intuitive to add a view and set its constraints inside the host. Having a height property and setting the constraint in the view is a great idea, but for future use, someone might use the view and unknowingly add a height constraint and that would cause conflicts.

> Or, if we want to avoid setting absolute sizes in the view at all, we might assign the button constraints proportionally?

Doing this would be a bit tricky, as we're using `UIButton.imageView`. Since we don't know how we'll use it in the future, I'd rather not do that now. We could end up deciding that we want to change the banner's height but keep the subviews the same size. Let me know if that makes sense!

Reply:
Sounds 100% good, and thank you for tackling this @hassaanelgarem !!